### PR TITLE
[skip-ci] GHA: Fix intermittent workflow error

### PR DESCRIPTION
**Pull request by cevich**
Periodically, the discussion-lock workflow throws the error: `Resource
not accessible by integration`

This was identified in the [upstream](https://github.com/dessant/lock-threads) issue 47, as caused by a version-5 change that adds support for management of discussions but requires additional permissions and possibly settings.  Given the low notification traffic from discussions, old discussions may remain valid for a long while, and are a useful community-interface:  Disable management of discussions.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago @lsm5 @mheon PTAL, this should fix the alerts we've been getting from this repo.  There's no practical way to test this PR, so it needs to be manually merged (the bot will not work).  If acceptable, I intend to open similar PRs for Buildah and Skopeo where we also receive this alert.

**Comment by cevich** [comment]
Clarification: "The fix worked for other people" is all I have to go on for this change.

**Review by Luap99 — CHANGES_REQUESTED** [review]
Let's please disable the thing for discussions instead, they are discussions and not meant locked just because someone closed them IMO.
https://github.com/dessant/lock-threads/issues/47#issuecomment-2041019276

And if we do want to lock them I guess we must configure it the same way we did for PRs/issues so that it does not use the defaults.

**Comment by cevich** [comment]
Thanks for your input Paul, sorry I probably should have tagged you as well.

IIRC, the original concern was over receiving e-mail notifications relating to old closed PRs/Issues.  So these would not be open/active discussions.

The workflow is already configured to use the same settings as for issues/PRs (and now discussions).

**Comment by Luap99** [comment]
> IIRC, the original concern was over receiving e-mail notifications relating to old closed PRs/Issues. So these would not be open/active discussions.

Yes exactly which is why this change is wrong because you enable locking of discussions, IMO discussions are from and for community and we see plenty of non maintainers active there as well, there is no requirement for us to interact with them.
Discussions are also sorted by on activity so if someone is active on a old discussion then it will be back at the font and be discoverable for others to join in.

> The workflow is already configured to use the same settings as for issues/PRs (and now discussions).

```
issue-inactive-days: '${{env.CLOSED_DAYS}}'
pr-inactive-days: '${{env.CLOSED_DAYS}}'
add-issue-labels: '${{env.LOCKED_LABEL}}'
add-pr-labels: '${{env.LOCKED_LABEL}}'
```

How do you set these for discussion? You currently don't so this fix would make things worse as it uses the defaults (which we do not liked)

**Comment by cevich** [comment]
Force-push: Disabled discussion-management, renamed job to better reflect this.

@Luap99 @mheon PTAL, this should be ready to go now.

File: .github/workflows/discussion_lock.yml
Comment (Luap99) [inline .github/workflows/discussion_lock.yml:42]: I think you should update the name on line 5 as well and maybe rename the file?

**Review by Luap99 — APPROVED** [review]
LGTM although I think if you change the name there you want to change it in the other place as well.

**Comment by cevich** [comment]
Force-push: Renamed job and workflow file.

**Comment by cevich** [comment]
> Looking forward to Thursday or Friday to see if this fixes the problem. Thank you!

Note: Both buildah and skopeo re-use this workflow via GHA magic. So by merging this we should find the problem is fixed everywhere.  Hopefully.

**Comment by cevich** [comment]
Oops, nope, spoke too soon.  The file rename is going to cause both Skopeo and Buildah to fail.

**Comment by edsantiago** [comment]
Good idea for a followup: add a huge all-caps warning saying "THIS FILE IS USED BY SKOPEO AND BUILDAH etc etc"

**Comment by cevich** [comment]
That is a good idea.

**Pull request merged**
